torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -510,3 +511,109 @@ def shard_parameter(
 
     # Now we can set the attribute appropriately.
     setattr(module, param_name, st)
+
+
+def init_sharded_tensor_from_local_result(
```

Comment:
This function doesn't seem to belong here, it is very specific to col wise sharding. Maybe we keep it in the linear implementation?

Reply:
ok, sure.